src/java.base/unix/classes/sun/nio/fs/UnixCopyFile.java

```diff
@@ -273,6 +276,18 @@ private static void copyFile(UnixPath source,
                                  long addressToPollForCancel)
         throws IOException
     {
+        boolean copied = false;
```

Comment:
The "copied" flag is a sign that the copyFile method needs to be refactor to split it into the copy of the contents and the file attributes. Also on first glance, re-opening both the source and destination files after a clone is a bug. We should (where possible) only open the source file for reading and the destination file for writing once.

Reply:
On macOS it's not possible to clone using file descriptors; paths are required.

Reply:
> On macOS it's not possible to clone using file descriptors; paths are required.

I understand that but what isn't clear is if the attributes are cloned too.  It might be that we only use it when not copying attributes.

Reply:
Based on my testing, attributes appear to be cloned by default except for the permission on linux-aarch64 and the (group) ownership on macosx-x64. If we want to suppress that behavior, then using clone would have to be made conditional.